File: spring-boot-iot-system/src/main/java/com/ghlzm/iot/system/entity/Organization.java
package com.ghlzm.iot.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * 组织机构实体
 */
@Data
@TableName("sys_organization")
public class Organization implements Serializable {
      private static final long serialVersionUID = 1L;

      /**
       * 主键
       */
      @TableId(value = "id", type = IdType.ASSIGN_ID)
      private Long id;

      /**
       * 租户ID
       */
      private Long tenantId;

      /**
       * 父ID
       */
      private Long parentId;

      /**
       * 组织名称
       */
      private String orgName;

      /**
       * 组织编码
       */
      private String orgCode;

      /**
       * 组织类型 dept/position/team
       */
      private String orgType;

      /**
       * 负责人ID
       */
      private Long leaderUserId;

      /**
       * 负责人姓名
       */
      private String leaderName;

      /**
       * 联系电话
       */
      private String phone;

      /**
       * 联系邮箱
       */
      private String email;

      /**
       * 状态 1启用 0禁用
       */
      private Integer status;

      /**
       * 排序
       */
      private Integer sortNo;

      /**
       * 备注
       */
      private String remark;

      /**
       * 创建人
       */
      private Long createBy;

      /**
       * 创建时间
       */
      private Date createTime;

      /**
       * 更新人
       */
      private Long updateBy;

      /**
       * 更新时间
       */
      private Date updateTime;

      /**
       * 删除标记
       */
      @TableLogic
      private Integer deleted;

      /**
       * 子组织列表（树形结构使用）
       */
      private List<Organization> children;
}

File: spring-boot-iot-ui/src/api/organization.ts
import { request } from './request'
import type { ApiEnvelope, IdType, PageResult } from '../types/api'

export interface Organization {
  id: IdType
  tenantId: IdType
  parentId: IdType
  orgName: string
  orgCode: string
  orgType: string
  leaderUserId?: IdType
  leaderName: string
  phone: string
  email: string
  status: number
  sortNo: number
  remark: string
  createBy: number
  createTime: string
  updateBy: number
  updateTime: string
  deleted: number
  children?: Organization[]
  hasChildren?: boolean
}

export interface OrganizationQueryParams {
  orgName?: string
  orgCode?: string
  status?: number
}

export interface OrganizationPageQueryParams extends OrganizationQueryParams {
  pageNum?: number
  pageSize?: number
}

function buildQuery(params: Record<string, unknown>) {
  const query = new URLSearchParams()
  Object.entries(params).forEach(([key, value]) => {
    if (value !== undefined && value !== null && value !== '') {
      query.append(key, String(value))
    }
  })
  return query.toString()
}

export const listOrganizations = (parentId?: IdType): Promise<ApiEnvelope<Organization[]>> => {
  const query = buildQuery({ parentId })
  return request<Organization[]>(`/api/organization/list${query ? `?${query}` : ''}`, { method: 'GET' })
}

export const pageOrganizations = (params: OrganizationPageQueryParams = {}): Promise<ApiEnvelope<PageResult<Organization>>> => {
  const query = buildQuery(params)
  return request<PageResult<Organization>>(`/api/organization/page${query ? `?${query}` : ''}`, { method: 'GET' })
}

export const listOrganizationTree = (): Promise<ApiEnvelope<Organization[]>> => {
  return request<Organization[]>('/api/organization/tree', { method: 'GET' })
}

export const getOrganization = (id: IdType): Promise<ApiEnvelope<Organization>> => {
  return request<Organization>(`/api/organization/${id}`, { method: 'GET' })
}

export const addOrganization = (data: Partial<Organization>): Promise<ApiEnvelope<Organization>> => {
  return request<Organization>('/api/organization', { method: 'POST', body: data })
}

export const updateOrganization = (data: Partial<Organization>): Promise<ApiEnvelope<Organization>> => {
  return request<Organization>('/api/organization', { method: 'PUT', body: data })
}

export const deleteOrganization = (id: IdType): Promise<ApiEnvelope<void>> => {
  return request<void>(`/api/organization/${id}`, { method: 'DELETE' })
}
